plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,268 @@
+	configValue := ""
+	exchangeName := ""
+	tradingPair := &model.TradingPair{Base: "XLM", Quote: "XLM"}
+	modes := []volumeFilterMode{volumeFilterModeExact, volumeFilterModeIgnore}
+	firstMarketID := MakeMarketID(exchangeName, "native", "native")
+
+	testCases := []struct {
+		name       string
+		marketIDs  []string
+		accountIDs []string
+		wantFilter *volumeFilter
+	}{
+		// TODO DS Confirm the empty config fails once validation is added to the constructor
+		{
+			name:       "1 market id",
+			marketIDs:  []string{"marketID"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 market ids",
+			marketIDs:  []string{"marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 dupe market ids, 1 distinct",
+			marketIDs:  []string{"marketID1", "marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "1 account id",
+			marketIDs:  []string{},
+			accountIDs: []string{"accountID"},
+		},
+		{
+			name:       "2 account ids",
+			marketIDs:  []string{},
+			accountIDs: []string{"accountID1", "accountID2"},
+		},
+		{
+			name:       "account and market ids",
+			marketIDs:  []string{"marketID"},
+			accountIDs: []string{"accountID"},
+		},
+	}
+
+	for _, k := range testCases {
+		// this lets us test both types of modes when varying the market and account ids
+		for _, m := range modes {
+			// this lets us test both constraints within the config
+			baseCapInBaseConfig := makeTestVolumeFilterConfig(1.0, -1.0, k.marketIDs, k.accountIDs, m)
+			baseCapInQuoteConfig := makeTestVolumeFilterConfig(-1.0, 1.0, k.marketIDs, k.accountIDs, m)
+
+			for _, config := range []*VolumeFilterConfig{baseCapInBaseConfig, baseCapInQuoteConfig} {
+				// configType is used to represent the type of config when printing test name
+				var configType string
+				if config.SellBaseAssetCapInBaseUnits != nil {
+					configType = "base"
+				} else {
+					configType = "quote"
+				}
+
+				// TODO DS Vary filter action between buy and sell, once buy logic is implemented.
+				wantFilter := makeWantVolumeFilter(config, firstMarketID, k.marketIDs, k.accountIDs, "sell")
+				t.Run(fmt.Sprintf("%s/%s/%s", k.name, configType, m), func(t *testing.T) {
+					actual, e := makeFilterVolume(
+						configValue,
+						exchangeName,
+						tradingPair,
+						testAssetDisplayFn,
+						utils.NativeAsset,
+						utils.NativeAsset,
+						&sql.DB{},
+						config,
+					)
+
+					if !assert.Nil(t, e) {
+						return
+					}
+
+					assert.Equal(t, wantFilter, actual)
+				})
+			}
+		}
+	}
+}
+
+func makeManageSellOffer(price, amount string) *txnbuild.ManageSellOffer {
+	if amount == "" {
+		return nil
+	}
+
+	return &txnbuild.ManageSellOffer{
+		Buying:  txnbuild.NativeAsset{},
+		Selling: txnbuild.NativeAsset{},
+		Price:   price,
+		Amount:  amount,
+	}
+}
+
+func TestVolumeFilterFn(t *testing.T) {
```

Comment:
reviewing later once we have the factory method well tested -- I think we should split these up into separate PRs